internal/sys/map_flags.go

```diff
@@ -0,0 +1,13 @@
+package sys
+
+import "github.com/cilium/ebpf/internal/unix"
+
+type MapFlags uint32
+
+const (
```

Comment:
This should be in `features` for now, and document that only these flags are supported as arguments to `HaveMapFlag`

Reply:
What I did is having it in `sys` (allowing usage in `sys` so no circular imports) and re-export it in `features`. Should I do it differently ?

Reply:
`type MapFlags` is fine, but the `const` block for the constants should be moved.

Reply:
(I think you could keep `MapFlags` in types.go, but it doesn't matter much.)

Reply:
Oh right, I moved the constants in the last commit and made the change so that `sys.MapFlags` is generated by `gentypes`. I'm not super happy if you have an improvement in mind that can help with this please don't hesitate